lib/derive-idol-err/src/lib.rs

```diff
@@ -28,25 +28,62 @@ pub fn derive(input: TokenStream) -> TokenStream {
         }
     };
 
-    // Assert that each variant is nonzero when cast to a `u32`; zero is
-    // reserved for success!
-    let variant_nonzero_assertions = data.variants.into_iter().map(|variant| {
-        let v = variant.ident;
-
-        // Inline version of
-        // ```
-        // static_assertions::const_assert_ne!(#ident::#v, 0)
-        // ```
-        quote! {
-            const _: [(); 0 - !{
-                const ASSERT: bool = #ident::#v as u32 != 0;
-                ASSERT
-            } as usize] = [];
+    let mut variant_errors = vec![];
+    let mut discriminant = None;
+    for v in &data.variants {
+        if v.fields != syn::Fields::Unit {
+            variant_errors.push(compile_error(
+                v.ident.span(),
+                "idol errors must be C-style enums",
+            ));
+        }
+        if let Some((_, d)) = &v.discriminant {
+            if let syn::Expr::Lit(syn::ExprLit {
+                lit: syn::Lit::Int(lit),
+                ..
+            }) = d
+            {
+                let value: i64 = match lit.base10_parse() {
+                    Ok(value) => value,
+                    Err(e) => {
+                        variant_errors.push(e.into_compile_error());
+                        continue;
+                    }
+                };
+                if value == 0 {
+                    variant_errors.push(compile_error(
+                        lit.span(),
+                        "error enums must not contain zero",
+                    ));
+                }
+                discriminant = Some(value);
+            } else {
+                variant_errors.push(
+                    compile_error(v.ident.span(), "idol errors must use simple positive integer discriminants"),
```

Comment:
I'm surprised rustfmt didn't shorten this for you

Reply:
Not only did it not shorten this, it's forcing it onto one line. I've managed to get it to stop by explicitly breaking the string across lines.